### PR TITLE
Fix integration tests for FSx for OpenZFS and Ontap

### PR DESCRIPTION
**Pull request by hanwen-pcluste**
See commit descriptions for details

### Checklist
- [ ] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [ ] Check all commits' messages are clear, describing what and why vs how.
- [ ] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [ ] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/integration-tests/tests/storage/test_fsx_lustre.py [hunk 1]
```diff
@@ -333,19 +333,26 @@ def test_multiple_fsx(
     bucket.upload_file(str(test_datadir / "s3_test_file"), "s3_test_file")
     import_path = "s3://{0}".format(bucket_name)
     export_path = "s3://{0}/export_dir".format(bucket_name)
+    partition = utils.get_arn_partition(region)
     num_new_fsx_lustre = 1
-    num_existing_fsx_ontap = 1
-    num_existing_fsx_open_zfs = 1
+    num_existing_fsx_ontap = 1 if partition != "aws-cn" else 0  # China does not have Ontap
```
Comment (gmarciani) [inline tests/integration-tests/tests/storage/test_fsx_lustre.py:338]: the file is named `test_fsx_lsutre.py`, but it also covers Ontap/OpenZFs. what about renaming it?

File: tests/integration-tests/tests/storage/test_fsx_lustre.py [hunk 2]
```diff
@@ -333,19 +333,26 @@ def test_multiple_fsx(
...
         # Only create more FSx when benchmarks are specified. Limiting OS to reduce cost of too many file systems
         num_existing_fsx = 50
     else:
         # Minimal total existing FSx is the number of Ontap and OpenZFS plus one existing FSx Lustre
         num_existing_fsx = num_existing_fsx_ontap + num_existing_fsx_open_zfs + 1
     num_existing_fsx_lustre = num_existing_fsx - num_existing_fsx_ontap - num_existing_fsx_open_zfs
-    mount_dirs = ["/shared"]  # OSU benchmark relies on /shared directory
-    for i in range(num_new_fsx_lustre + num_existing_fsx - 1):
-        mount_dirs.append(f"/fsx_mount_dir{i}")
+    fsx_lustre_mount_dirs = ["/shared"]  # OSU benchmark relies on /shared directory
```
Comment (gmarciani) [inline tests/integration-tests/tests/storage/test_fsx_lustre.py:347]: So, we are not using the additional file systems for OSU benchmarks. Is that the expected outcome or do we want OSU to use also Ontap/OpenZFS?

**Review by gmarciani — APPROVED** [review]
Approved with minor comments.

File: tests/integration-tests/tests/storage/test_fsx_lustre.py (hunk 1 above)
Comment (hanwen-pcluste) [reply tests/integration-tests/tests/storage/test_fsx_lustre.py:338]: Good catch. I will rename the tests in another PR

File: tests/integration-tests/tests/storage/test_fsx_lustre.py (hunk 2 above)
Comment (hanwen-pcluste) [reply tests/integration-tests/tests/storage/test_fsx_lustre.py:347]: Yes. we can improve the test in the future

**Pull request merged**